lib/improver/blending/weights.py

```diff
@@ -244,6 +248,538 @@ def process_coord(cube, coordinate, coord_exp_vals=None,
+            weights_len = (
+                len(self.config_dict[key][self.weights_coord_name]))
+            if weighting_len != weights_len:
+                msg = ("{} is {}, {} is {}."
+                       "These items in the configuration dictionary "
+                       "have different lengths i.e. {} != {}".format(
+                           self.weighting_coord_name,
+                           self.config_dict[key][self.weighting_coord_name],
+                           self.weights_coord_name,
+                           self.config_dict[key][self.weights_coord_name],
+                           weighting_len, weights_len))
+                raise ValueError(msg)
+
+    def _arrange_interpolation_inputs(self, cube):
+        """Organise the inputs required for the linear interpolation.
+
+        Args:
+            cube (iris.cube.Cube):
+                Cube containing the diagnostic that will be interpolated.
+
+        Returns:
+            (tuple): tuple containing
+
+                **source_points** (np.ndarray):
+                    Points within the configuration dictionary that will
+                    be used as the input to the interpolation.
+
+                **target_points** (np.ndarray):
+                    Points within the cube that will be the target points
+                    for the interpolation.
+
+                **associated_data** (np.ndarray):
+                    Weights from the configuration dictionary that will be
+                    used as the input to the interpolation.
+
+                **fill_value** (tuple):
+                    Values that be used if extrapolation is required. The
+                    fill values will be used as constants that are extrapolated
+                    if the target_points are outside the source_points
+                    provided.
+
+        """
+        config_point, = cube.coord(self.config_coord_name).points
+        source_points = (
+            self.config_dict[config_point][self.weighting_coord_name])
+        source_points = np.array(source_points)
+        if "units" in self.config_dict[config_point].keys():
+            units = cf_units.Unit(self.config_dict[config_point]["units"])
+            source_points = units.convert(
+                source_points, cube.coord(self.weighting_coord_name).units)
+
+        target_points = cube.coord(self.weighting_coord_name).points
+        associated_data = (
+            self.config_dict[config_point][self.weights_coord_name])
+
+        fill_value = (associated_data[0], associated_data[-1])
+        return source_points, target_points, associated_data, fill_value
+
+    def _create_new_weights_cube(self, cube, weights):
+        """Create a cube to contain the output of the interpolation.
+
+        Args:
+            cube (iris.cube.Cube):
+                Cube containing the diagnostic that was used for the
+                interpolation.
```

Comment:
Again - I think the cube is used for blending, not interpolation, although I'm a little confused.

Reply:
Yes, I've corrected this, as we're not actually interpolating this cube. We are just using some information from the cube to enable us to do some interpolation.